maestro-cli/src/main/java/maestro/cli/device/DeviceService.kt

```diff
@@ -76,10 +76,13 @@ object DeviceService {
                     PrintUtils.message("Setting the device locale to ${device.language}_${device.country}...")
                     val driver = AndroidDriver(dadb)
                     driver.open()
-                    driver.setDeviceLocale(
+                    val result = driver.setDeviceLocale(
                         country = device.country,
                         language = device.language
                     )
+                    if (result != -1) {
+                        throw IllegalStateException("Android failed to update a device locale")
```

Comment:
This is not debuggable really I think we should have failure reasons here and throw proper errors with message

Reply:
Also how the failure in setting locale works on iOS?

Reply:
+1 on what Amanjeet said, we should have as descriptive error messages as possible and avoid these catch all messages if we can. Do we know what can go wrong here? If yes, can we categorize it? If no, let's maybe display it to the user?

Reply:
added more helpful failure reasons for Android and made exception message more descriptive

Reply:
> Also how the failure in setting locale works on iOS?

There is sadly no way to verify that `xcrun simctl spawn defaults write` will always be successful. I think good documentation and exception messages are enough here for the users to be able to debug the reason